tests/system/test_system.py

```diff
@@ -1814,12 +1846,116 @@ def test_count_query_in_transaction(client, cleanup):
 
     with pytest.raises(ValueError) as exc:
         create_in_transaction_helper(transaction, client, collection_id, cleanup)
-        assert exc.exc_info == "Collection can't have more than 2 documents"
+    assert str(exc.value) == "Collection can't have more than 2 docs"
 
     collection = client.collection(collection_id)
 
-    query = collection.where("a", "==", 1)
+    query = collection.where(filter=FieldFilter("a", "==", 1))
     count_query = query.count()
     result = count_query.get()
     for r in result[0]:
         assert r.value == 2  # there are still only 2 docs
+
+
+def test_query_with_and_composite_filter(query_docs):
+    collection, stored, allowed_vals = query_docs
+    and_filter = And(
+        filters=[
+            FieldFilter("stats.product", ">", 5),
+            FieldFilter("stats.product", "<", 10),
+        ]
+    )
+
+    query = collection.where(filter=and_filter)
+    for result in query.stream():
+        assert result.get("stats.product") > 5
+        assert result.get("stats.product") < 10
+
+
+def test_query_with_or_composite_filter(query_docs):
+    collection, stored, allowed_vals = query_docs
+    or_filter = Or(
+        filters=[
+            FieldFilter("stats.product", ">", 5),
+            FieldFilter("stats.product", "<", 10),
+        ]
+    )
+    query = collection.where(filter=or_filter)
+    gt_5 = 0
+    lt_10 = 0
+    for result in query.stream():
+        value = result.get("stats.product")
+        assert value > 5 or value < 10
+        if value > 5:
+            gt_5 += 1
+        if value < 10:
+            lt_10 += 1
+
+    assert gt_5 > 0
+    assert lt_10 > 0
+
+
+def test_query_with_complex_composite_filter(query_docs):
+    collection, stored, allowed_vals = query_docs
+    field_filter = FieldFilter("b", "==", 0)
+    or_filter = Or(
+        filters=[FieldFilter("stats.sum", "==", 0), FieldFilter("stats.sum", "==", 4)]
+    )
+    query = collection.where(filter=field_filter).where(filter=or_filter)
+
+    sum_0 = 0
+    sum_4 = 0
+    for result in query.stream():
+        assert result.get("b") == 0
+        assert result.get("stats.sum") == 0 or result.get("stats.sum") == 4
+        if result.get("stats.sum") == 0:
+            sum_0 += 1
+        if result.get("stats.sum") == 4:
+            sum_4 += 1
+
+    assert sum_0 > 0
+    assert sum_4 > 0
+
+
+def test_or_query_in_transaction(client, cleanup):
```

Comment:
looking for a few more nested test examples in all cases like: `a ==1 || (b > 5 && a == 2)`, `a == 1 && (b > 5 || b < 2)`

Reply:
Added some more complex filtering above under `test_query_with_complex_composite_filter`.

I could not add more complex filter test using transaction because it would require adding index for each of the UNIQUE_RESOURCE_ID (different one each time I run the test, meaning new index for each time the test is run).

So I simply added running the test with complex And/or without transaction in `test_query_with_complex_composite_filter`.

I added the same test to the async test as well.